lib/src/facts/linux/networking_resolver.cc

```diff
@@ -184,4 +213,45 @@ namespace facter { namespace facts { namespace linux {
             }
         }
     }
+
+    string networking_resolver::get_bond_master(const std::string& name) const {
+        static bool have_logged_about_bonding = false;
+        auto ip_command = lth_exe::which("ip");
+        if (ip_command.empty()) {
+            if (!have_logged_about_bonding) {
+                 LOG_DEBUG("Could not find the 'ip' command. Physical macaddress for bonded interfaces will be incorrect.");
+                 have_logged_about_bonding = true;
+            }
+            return {};
+        }
+
+        string bonding_master;
+
+        lth_exe::each_line(ip_command, {"link", "show", name}, [&bonding_master](string& line) {
+            if (line.find("SLAVE") != string::npos) {
+                vector<boost::iterator_range<string::iterator>> parts;
+                boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+                // We have to use find_if here since a boost::iterator_range doesn't compare properly to a string.
+                auto master = find_if(parts.begin(), parts.end(), [](boost::iterator_range<string::iterator>& part){
+                    string p {part.begin(), part.end()};
```

Comment:
I tried a few variations on that and couldn't get it to work - I think a `const char*` is *just* different enough from a real iterator range to break `equals`, despite pointers being nominally "iterators"